### ISSUE TITLE
Bug: Time Advance Stuck

### ISSUE THREAD

**Issue by Wargamer777** [MegaMek/mekhq#2386]
### Environment
 .47.16
Windows 10

### Description
When clicking the Advance Day button, the day doesn't advance along the top of the screen.

The Daily Activity Log does show a day advance but only appends the log directly below the previous day.

[BHM Day Advance Problem.cpnx.gz](https://github.com/MegaMek/mekhq/files/5846293/BHM.Day.Advance.Problem.cpnx.gz)

**Comment by Windchild292** [comment]
This is first on my list for the morning. Can I get the contents of the /logs folder and any customs please before then.

**Comment by Wargamer777** [comment]
Attached zipped logs folder.  No particular customs.

[logs.zip](https://github.com/MegaMek/mekhq/files/5846594/logs.zip)

**Comment by Wargamer777** [comment]
I spoke too soon..  1 Custom attached.


[Orion ON1-H Mk II.zip](https://github.com/MegaMek/mekhq/files/5846603/Orion.ON1-H.Mk.II.zip)

**Comment by sixlettervariables** [comment]
@Wargamer777 we need mekhqlog.txt and megameklog.txt

**Comment by Wargamer777** [comment]
> 
> 
> @Wargamer777 we need mekhqlog.txt and megameklog.txt

That's attached above in the logs.zip file.

**Comment by BowtieBob** [comment]
This might be the same issue I was having with the heat sink replacement using aero heat sinks. After the replacement, the day at the top of the screen stops advancing.

**Comment by sixlettervariables** [comment]
I only see:
![image](https://user-images.githubusercontent.com/8238690/105278133-def61280-5b72-11eb-85eb-ddb29cdaba05.png)

**Comment by Wargamer777** [comment]
Ok, that was weird.  Here's the two logs:

[mekhqlog.txt](https://github.com/MegaMek/mekhq/files/5848784/mekhqlog.txt)
[megameklog.txt](https://github.com/MegaMek/mekhq/files/5848785/megameklog.txt)

**Comment by sixlettervariables** [comment]
```
Exception in thread "AWT-EventQueue-0" java.lang.ClassCastException: mekhq.campaign.parts.AeroHeatSink cannot be cast to mekhq.campaign.parts.equipment.EquipmentPart
	at mekhq.campaign.parts.equipment.MissingEquipmentPart.fix(MissingEquipmentPart.java:177)
	at mekhq.campaign.unit.actions.RestoreUnitAction.execute(RestoreUnitAction.java:55)
	at mekhq.gui.adapter.UnitTableMouseAdapter.actionPerformed(UnitTableMouseAdapter.java:478)
...
```

Duplicate of #2365.

Fixed in #2366.

**Comment by sixlettervariables** [comment]
Try this:
[BHM.Day.Advance.Problem.cpnx.gz](https://github.com/MegaMek/mekhq/files/5849828/BHM.Day.Advance.Problem.cpnx.gz)

My apologies but for the time being AeroHeatSinks will block your repairs.
